### PR TITLE
Add javadoc.

### PR DESCRIPTION
**Pull request by ghislainfourny**
Javadoc for the top-level node of the intermediate representation (renaming ExpressionOrClause to Node).

Also renaming getDescendants() to getChildren().

git doesn't seem to have seen that this is a rename and shows as deletion+addition.

### CODE REVIEW

**Comment by CanBerker** [comment]
> git doesn't seem to have seen that this is a rename and shows as deletion+addition.

For future references, using "git mv" command to rename a file usually fixed the issue you have mentioned. Hope it helps!

File: src/main/java/org/rumbledb/expressions/Node.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    public abstract String serializationString(boolean prefix);
+
+    /**
+     * Returns all children nodes.
+     * 
+     * @return the children nodes as a list.
+     */
+    public final List<Node> getChildren() {
+        return getDescendants(false);
+    }
```
Comment (CanBerker) [inline src/main/java/org/rumbledb/expressions/Node.java]: Perfect

File: src/main/java/sparksoniq/semantics/visitor/AbstractExpressionOrClauseVisitor.java [hunk 2]
```diff
@@ -75,19 +75,19 @@
 
 public abstract class AbstractExpressionOrClauseVisitor<T> {
 
-    public T visit(ExpressionOrClause expression, T argument) {
+    public T visit(Node expression, T argument) {
```
Comment (CanBerker) [inline src/main/java/sparksoniq/semantics/visitor/AbstractExpressionOrClauseVisitor.java]: I have a feeling "Node expression" naming can be confusing down the line as we have an Expression class as well. How about naming the arguments as "Node node" in all similar places where we have refactored from ExpressionOrClause

File: src/main/java/sparksoniq/semantics/visitor/AbstractExpressionOrClauseVisitor.java [hunk 3]
```diff
@@ -75,19 +75,19 @@
 
 public abstract class AbstractExpressionOrClauseVisitor<T> {
```
Comment (CanBerker) [inline src/main/java/sparksoniq/semantics/visitor/AbstractExpressionOrClauseVisitor.java]: AbstractNodeVisitor?

**Comment by ghislainfourny** [comment]
Very good comments. Thank you!

File: src/main/java/sparksoniq/semantics/visitor/RuntimeIteratorVisitor.java [hunk 4]
```diff
@@ -145,17 +145,17 @@
 import org.rumbledb.expressions.quantifiers.QuantifiedExpression;
 import org.rumbledb.expressions.quantifiers.QuantifiedExpressionVar;
 
-public class RuntimeIteratorVisitor extends AbstractExpressionOrClauseVisitor<RuntimeIterator> {
+public class RuntimeIteratorVisitor extends AbstractNodeVisitor<RuntimeIterator> {
 
     @Override
-    public RuntimeIterator visit(ExpressionOrClause expression, RuntimeIterator argument) {
+    public RuntimeIterator visit(Node expression, RuntimeIterator argument) {
```
Comment (CanBerker) [inline src/main/java/sparksoniq/semantics/visitor/RuntimeIteratorVisitor.java]: Should we also opt for 'node' variable name here as well or do you think expression is better fitting ?

File: src/main/java/sparksoniq/semantics/visitor/RuntimeIteratorVisitor.java [hunk 5]
```diff
@@ -145,17 +145,17 @@
...
         return expression.accept(this, argument);
     }
 
     @Override
-    public RuntimeIterator visitDescendants(ExpressionOrClause expression, RuntimeIterator argument) {
+    public RuntimeIterator visitDescendants(Node expression, RuntimeIterator argument) {
```
Comment (CanBerker) [inline src/main/java/sparksoniq/semantics/visitor/RuntimeIteratorVisitor.java]: node

File: src/main/java/sparksoniq/semantics/visitor/StaticContextVisitor.java [hunk 6]
```diff
@@ -89,15 +89,15 @@ public void setConfigForFinalPass() {
     }
 
     @Override
-    protected StaticContext defaultAction(ExpressionOrClause expression, StaticContext argument) {
+    protected StaticContext defaultAction(Node expression, StaticContext argument) {
```
Comment (CanBerker) [inline src/main/java/sparksoniq/semantics/visitor/StaticContextVisitor.java]: node

File: src/main/java/sparksoniq/semantics/visitor/StaticContextVisitor.java [hunk 7]
```diff
@@ -89,15 +89,15 @@ public void setConfigForFinalPass() {
...
         StaticContext generatedContext = visitDescendants(expression, argument);
         // initialize execution mode by visiting children and expressions first, then calling initialize methods
         expression.initHighestExecutionMode();
         return generatedContext;
     }
 
     @Override
-    public StaticContext visit(ExpressionOrClause expression, StaticContext argument) {
+    public StaticContext visit(Node expression, StaticContext argument) {
```
Comment (CanBerker) [inline src/main/java/sparksoniq/semantics/visitor/StaticContextVisitor.java]: node

File: src/main/java/sparksoniq/semantics/visitor/VisitorHelpers.java [hunk 8]
```diff
@@ -12,11 +12,11 @@
 
 public class VisitorHelpers {
 
-    public static RuntimeIterator generateRuntimeIterator(ExpressionOrClause expression) {
+    public static RuntimeIterator generateRuntimeIterator(Node expression) {
         return new RuntimeIteratorVisitor().visit(expression, null);
     }
 
-    public static void populateStaticContext(ExpressionOrClause expression) {
+    public static void populateStaticContext(Node expression) {
```
Comment (CanBerker) [inline src/main/java/sparksoniq/semantics/visitor/VisitorHelpers.java]: node

**Comment by ghislainfourny** [comment]
Very good catches. This is now fixed.

**Pull request merged**